### PR TITLE
Various improvements involving cursor descriptions

### PR DESCRIPTION
**Pull request by johnanthonyowens**

### CODE REVIEW

File: src/ctds/cursor.c [hunk 1]
```diff
@@ -469,49 +491,61 @@ static PyObject* create_column_description(const struct Column* column)
...
+
+            if (PyErr_Occurred())
+            {
+                Py_DECREF(tuple);
+                tuple = NULL;
             }
             else
             {
-                break;
+                description->_obj = tuple;
```
Comment (joshuahlang) [inline src/ctds/cursor.c:522]: I would set `tuple = NULL` and add a comment that the reference was moved to `description->_obj`

File: tests/test_cursor_description.py [hunk 2]
```diff
@@ -167,13 +168,76 @@ def test_types(self):
                 else: # pragma: nocover
                     pass
 
+    def test_multiple_descriptions(self):
+        with self.connect() as connection:
```
Comment (joshuahlang) [inline tests/test_cursor_description.py:172]: in `TestCursorCallproc` there's a `stored_procedure` function to create/drop a sproc. Probably should factor that out and use it here. Also that function itself appears to need a `try: except:` around the `yield`

File: src/ctds/cursor.c (hunk 1 above)
Comment (johnanthonyowens) [reply src/ctds/cursor.c:522]: Done.

File: tests/test_cursor_description.py (hunk 2 above)
Comment (johnanthonyowens) [reply tests/test_cursor_description.py:172]: Done.

**Pull request closed without merging**